src/mlpack/methods/ann/layer/c_relu_impl.hpp

```diff
@@ -82,11 +82,15 @@ void CReLUType<MatType>::Forward(
 
 template<typename MatType>
 void CReLUType<MatType>::Backward(
-    const MatType& input, const MatType& gy, MatType& g)
+    const MatType& input,
+    const MatType& output,
+    const MatType& gy,
+    MatType& g)
 {
-  MatType temp = gy % (input >= 0.0);
-  g = temp.rows(0, (input.n_rows / 2 - 1)) - temp.rows(input.n_rows / 2,
-      (input.n_rows - 1));
+  // I believe this needs to be > 0.0, not >= 0.0
```

Comment:
True, but in this case you are picking from two distinct output values for your backprop depending on which one you choose (> or >=).  I think it might possibly be more relevant here than the others, but not sure if it actually matters.

Reply:
Yeah, agreed, `g[i]` is not guaranteed to be the same in this case because `gy[i]` may not be `-gy[2i]`.  I'm open to changing to `>`, but I haven't developed any justification for doing so.  In the absence of a justification I'd leave as-is just to avoid changing behavior, but if you do have a situation where `>` actually would be better to use than `>=`, I'm alright with the change. :+1: Your call, it's a bit tangential to the purpose of the PR itself.

Reply:
I committed a new version defining the gradient in terms of the inputs, rather than the outputs.  I think this is the correct way to do it.  I used >=, since that is what you went with in the others.

Reply:
Thanks, that sounds good to me. :+1: